### ISSUE TITLE
Opinion: Arguments should not be required

### ISSUE THREAD

**Issue by Rowlando13** [pallets/click#2824]
Docs: 
- Arguments: 
It is possible to make an argument required by setting required=True. It is not recommended since we think command line tools should gracefully degrade into becoming no ops. We think this because command line tools are often invoked with wildcard inputs and they should not error out if the wildcard is empty.

Reasoning: 
Beyond the reasoning stated, I don't know of any.

Possible features: N/A

**Comment by davidism** [comment]
I don't think this one is a problem. As it says, the feature is already supported (`nargs=-1, required=True`). Stating why that might be a bad idea if you're trying to design a CLI tool that behaves well in pipelines seems fine to me.
